src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs

```diff
@@ -261,11 +261,6 @@ public static void AddProviderTransparent(TypeDescriptionProvider provider, obje
         /// </summary>
         private static void CheckDefaultProvider(Type type)
         {
-            if (s_defaultProviders.ContainsKey(type))
-            {
-                return;
-            }
-
             lock (s_internalSyncObject)
```

Comment:
Can the call to `ContainsKey` be outside the `lock` by using the double-checked lock pattern? That would help performance since it avoids the `lock` for the majority of cases.

Reply:
@steveharter , no, it can not. Bug was linked exactly with double-checked lock pattern, which was applied before. Yes, I understand, that perfomance of my decision will be worse, that's why I suggested using ConcurrentDictionary.

Reply:
@steveharter , you are right, it's my mistake. I had just fixed it by https://github.com/Maximys/runtime/commit/0f46857052d699accf7277d709e74e52d98030f2 commit